openfisca_core/holders/holder.py

```diff
@@ -63,43 +125,64 @@ def clone(self, population):
 
         return new
 
-    def create_disk_storage(self, directory=None, preserve=False):
+    def create_disk_repo(
+        self,
+        directory: str | None = None,
+        preserve: bool = False,
+    ) -> Storage:
         if directory is None:
             directory = self.simulation.data_storage_dir
-        storage_dir = os.path.join(directory, self.variable.name)
+
+        storage_dir = os.path.join(directory, self.name)
+
         if not os.path.isdir(storage_dir):
             os.mkdir(storage_dir)
-        return storage.OnDiskStorage(
-            storage_dir, self._eternal, preserve_storage_dir=preserve
-        )
 
-    def delete_arrays(self, period=None):
+        return DiskRepo(storage_dir, preserve)
+
+    def delete_arrays(self, period: Any = None) -> None:
         """
         If ``period`` is ``None``, remove all known values of the variable.
 
         If ``period`` is not ``None``, only remove all values for any period included in period (e.g. if period is "2017", values for "2017-01", "2017-07", etc. would be removed)
         """
 
-        self._memory_storage.delete(period)
-        if self._disk_storage:
-            self._disk_storage.delete(period)
+        if self.eternal and period is not None:
+            period = periods.period(periods.DateUnit.ETERNITY)
+
+        else:
+            period = periods.period(period)
+
+        for store in self.stores.values():
+            store.delete(period)
 
-    def get_array(self, period):
+        return None
+
+    def get_array(self, period: Period) -> numpy.ndarray | None:
         """
-        Get the value of the variable for the given period.
+        Gets the value of the variable for the given period.
 
         If the value is not known, return ``None``.
         """
-        if self.variable.is_neutralized:
-            return self.default_array()
-        value = self._memory_storage.get(period)
-        if value is not None:
-            return value
-        if self._disk_storage:
-            return self._disk_storage.get(period)
+        if self.neutralised:
+            return self.variable.default_array(self.population.count)
+
+        if self.eternal:
+            period = periods.period(periods.DateUnit.ETERNITY)
+
+        else:
+            period = periods.period(period)
+
+        for store in self.stores.values():
+            value = store.get(period)
+
+            if value is not None:
+                return value
+
+        return None
 
     def get_memory_usage(self) -> MemoryUsage:
-        """Get data about the virtual memory usage of the Holder.
+        """Gets data about the virtual memory usage of the Holder.
```

Comment:
I would use [imperative](https://stackoverflow.com/questions/59902102/why-is-imperative-mood-important-for-docstrings) here and in other places